src/components/editor/Bindings/OnIncompatibleSchemaChange/useUpdateOnIncompatibleSchemaChange.ts

```diff
@@ -0,0 +1,129 @@
+import { modifyDraftSpec } from 'api/draftSpecs';
+import { useEntityType } from 'context/EntityContext';
+import { useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import { BASE_ERROR } from 'services/supabase';
+import { useBinding_bindings } from 'stores/Binding/hooks';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+import { getBindingIndex } from 'utils/workflow-utils';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_mutate,
+} from '../../Store/hooks';
+import { AutoCompleteOption } from './types';
+
+export interface BindingMetadata {
+    bindingIndex: number;
+    collection: string;
+}
+
+const updateSchema = (binding: any, newVal: any) => {
+    if (newVal) {
+        binding.onIncompatibleSchemaChange = newVal;
+    } else {
+        delete binding.onIncompatibleSchemaChange;
+    }
+};
+
+function useUpdateOnIncompatibleSchemaChange() {
+    const intl = useIntl();
+    const entityType = useEntityType();
+
+    // Binding Store
+    const bindings = useBinding_bindings();
+
+    // Draft Editor Store
+    const draftId = useEditorStore_persistedDraftId();
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    const updateOnIncompatibleSchemaChange = useCallback(
+        async (
+            newVal: AutoCompleteOption['val'] | undefined,
+            bindingMetadata: BindingMetadata[]
+        ) => {
+            const bindingMetadataExists = hasLength(bindingMetadata);
+
+            const invalidBindingIndex = bindingMetadataExists
+                ? bindingMetadata.findIndex(
+                      ({ bindingIndex }) => bindingIndex === -1
+                  )
+                : -1;
+
+            if (
+                !draftId ||
+                !mutateDraftSpecs ||
+                draftSpecs.length === 0 ||
+                invalidBindingIndex > -1
+            ) {
+                // TODO (onschema) update message
+                const errorMessageId = bindingMetadataExists
+                    ? 'workflows.collectionSelector.manualBackfill.error.message.singleCollection'
+                    : 'workflows.collectionSelector.manualBackfill.error.message.allBindings';
```

Comment:
Haha I wondered why the code in this file felt so familiar. Regarding the TODO comment, would you like to update these message IDs now or later?

Reply:
Later on... would like to make a lot of the stuff that's duplicated between all these editors.